src/util/png_image.hpp

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include "files.hpp"
+
+#include <png.h>
+
+#include <filesystem>
+
+namespace image {
+
+template <class T>
+void
```

Comment:
```suggestion
bool
```

return false on error